collection-scripts/gather_network_logs

```diff
@@ -17,7 +17,28 @@ function gather_multus_data {
 CLUSTER_NODES="${@:-$(oc get node -l node-role.kubernetes.io/master -oname)}"
 /usr/bin/gather_multus_logs $CLUSTER_NODES
 
-function gather_ovn_kubernetes_data {
+function gather_ovn_kubernetes_data_interconnect_mode {
+  echo "INFO: Gathering ovn-kubernetes DBs"
+  OVNKUBE_CONTROLLER_PODS=($(oc -n openshift-ovn-kubernetes get pods -l app=ovnkube-node -o=jsonpath='{.items[*].metadata.name}'))
+  # collect dbs from each node
```

Comment:
If we have 500 nodes do we really want data from all 500? Should this be limited in some way (or an extra script, where in a user specifies the nodes they want to collect this data from?

I worry this will take forever to collect on large clusters, and worry further that we will be collecting a large set of data no one will fully review.

Reply:
> If we have 500 nodes do we really want data from all 500? Should this be limited in some way 

We could limit it to 100 nodes when using this main script.,perhaps?  Note that DB size on each node is going to be smaller than the one giant DB we collect today from the leader node because we are splitting the data across to each node locally rather than keeping it centralized.

> (or an extra script, where in a user specifies the nodes they want to collect this data from?

if we want to do extra script, that is fine - infact we could just opt-into `gather-dbs-from-all-nodes` script as add-on which is how it was before the 4.10 release

however the problem with capturing from specific DBs is that when users open bugs, they need to know which dbs to capture from and this is not trivial to determine, its possible pod moves between nodes and we might not know which node it was on when must-gather was triggered for collection. Since MG is a real-time event we mostly always need the full set of dbs across all nodes unless the user knows all the variables exactly (where was the pod, endpointslice, service) at the time if gathering.

> 
> I worry this will take forever to collect on large clusters, and worry further that we will be collecting a large set of data no one will fully review.

QQ: Does MG as of today actually scale all the way to 500 nodes? I usually see cases gathering sos-reports instead of a full must-gather from such large clusters... or inspect logs from specific namespaces usually...when the numbers of nodes is that high.

Reply:
I will discuss with the wider networking team and come back to this.

Reply:
> however the problem with capturing from specific DBs is that when users open bugs, they need to know which dbs to capture from and this is not trivial to determine, 

Let me clarify this more, pod is the simplest unit of repersentation and while it might be easy for us to say which ndoe the pod is on currently that is facing issues just before running a must-gather, what I meant was we have several dimentions of features in networking - egressIPs/egressFWs/network-policies/egressGWs which also have other associated hop routes etc configured on each node for network plumbing and hence depending on which feature we are debugging it might become crucial to gather all DBs.

I think as of now I see 3 options:
 
Option1 : remove DB collection as part of default MG and add a `gather-ovnk-dbs` script that will be opt-in -> this is always gather dbs from all nodes
Option2: do the above + take a node argument for specific node collections -> can be enhancement in future
Option3: Limit collecting DBs in the original script by default upto 100 nodes and then use gather-ovnk-dbs for large clusters -> this is confusing for users. I think we should stay consistent and go with option1.

Reply:
We spoke about this in the team meeting. Everyone unitedly voted for collecting DBs by default and see if we can go that way before we need to pivot.

So here is what we are planning to do -> run the gathering in parallel across all nodes and tar them together as single entity instead of smaller tar's. 

@sferich888 : Do you have max limits for 500 node clusters - both time and size wise for a single MG tar file that I can use as values to compare a run against? That data can immensely help me versus running in blind with some numbers...

Reply:
Hi @tssurya,
Eric (sferich888) is on PTO today.  If he doesn't have any further objections upon his return, I'll be glad to provide 'approve'.
I'll wait until he returns so I don't approve now and lose the opportunity for further refinement (if any exists).
Thanks,
Rick